price/src/price_service.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{price_generator::MessageReceiver, GrpcResult};
+use helium_proto::services::price_oracle::{PriceOracle, PriceOracleReportV1, PriceOracleReqV1};
+use tonic::Request;
+
+pub struct PriceService {
+    price_watch: MessageReceiver,
+}
+
+impl PriceService {
+    pub fn new(price_watch: MessageReceiver) -> anyhow::Result<Self> {
+        Ok(Self { price_watch })
+    }
+}
```

Comment:
This is not required I think, the price generator itself keeps track of the last price fetched, in case of an rpc failure we just update the timestamp and return that.